shell/platform/linux/fl_renderer_egl.cc

```diff
@@ -0,0 +1,182 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "fl_renderer_egl.h"
+
+#include <EGL/egl.h>
+#include <EGL/eglext.h>
+#include <GLES2/gl2.h>
+
+#include "flutter/shell/platform/embedder/embedder.h"
+
+typedef struct {
+  EGLint egl_major;
+  EGLint egl_minor;
+
+  EGLDisplay egl_display;
+  EGLSurface egl_surface;
+  EGLContext egl_context;
+} FlRendererEGLPrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE(FlRendererEGL,
+                           fl_renderer_egl,
+                           fl_renderer_get_type())
+
+static gboolean fl_renderer_egl_start(FlRenderer* engine, GError** error) {
+  FlRendererEGL* self = FL_RENDERER_EGL(engine);
+  FlRendererEGLPrivate* priv = static_cast<FlRendererEGLPrivate*>(
+      fl_renderer_egl_get_instance_private(self));
+
+  /* Note the use of EGL_DEFAULT_DISPLAY rather than sharing an existing display
```

Comment:
Style nit, missed this before: while Google style doesn't strictly require it, `//` for inline comments is vastly more common in Google-styled code, including the engine, so please use that here and for other inline comments. `/* */` for the function-level comments is fine.